utility/TypeChecker.java

```diff
@@ -0,0 +1,79 @@
+package gov.nih.nci.bento.utility;
+
+import java.util.List;
+import java.util.Map;
+
+public class TypeChecker {
+    /**
+     * A generalized type checking method that can verify various collection types and their nested elements.
+     *
+     * @param obj The object to check
+     * @param containerType The expected container type (e.g., List.class, Map.class)
+     * @param elementTypes The expected element types in order:
+     *                     - For List<E>: pass the element type
+     *                     - For Map<K,V>: pass key type then value type, or Object.class for any type
+     * @return true if the object matches the expected types, false otherwise
+     */
+    public static boolean isOfType(Object obj, Class<?> containerType, Class<?>... elementTypes) {
```

Comment:
Eg: isOfType(obj, Map.class, String.class, Map.class, String.class, Integer.class) checks for the type Map<String, Map\<String, Integer\>>